v2/blockstore/readwrite.go

```diff
@@ -125,6 +130,10 @@ func (b *ReadWrite) isFinalized() bool {
 	return b.header.CarV1Size != 0
 }
 
+// TODO
+// 1. Can we consider finalize to be the `Close` of the read write blockstore ?
+// 2. Are Finalize/close idempotent ? Can we call them multiple times without side-effects/errors ?
```

Comment:
1) Finalize is not really a Close. See https://github.com/ipld/go-car/pull/114, which makes this much clearer.

2) They are not idempotent. At the current wip-v2 HEAD, calling Finalize a second time will panic, and calling ReadOnly.Close a second time will call os.File.Close a second time, which itself is documented to error on repeated calls.

Is there a particular reason you want idempotency? You could always accomplish it via something like boolean fields or https://golang.org/pkg/sync/#Once.

Reply:
On 1, we've agreed to have Finalize be the read-write blockstore's Close.

On 2, we agreed that they don't need to be idempotent.